src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -157,30 +157,144 @@ public bool TryGetPackageVersion(PackageInfo packageInfo, out SemanticVersion ve
+                {
+                    dirtyFiles.Add(componentFilePath);
+                }
             }
         }
+
+#endregion
+
+        #region Private Methods
+
+        private static void PopulateComponentJsonMaps(string mainComponentsJsonPath, string enlistmentRoot)
+        {
+            var mainComponentsJsonDocument = GetJsonDocumentForComponentsFile(mainComponentsJsonPath);
+
+            if (mainComponentsJsonDocument != null)
+            {
+                ComponentFileToDocumentMap[mainComponentsJsonPath] = mainComponentsJsonDocument;
+                PopulateComponentToFileMapForFile(mainComponentsJsonDocument, mainComponentsJsonPath);
+
+                // Process sub components.json
+                var imports = mainComponentsJsonDocument["Imports"];
+                if (imports != null)
+                {
+                    foreach (var import in imports)
+                    {
+                        string subComponentFileName = (string)import;
+
+                        if (!string.IsNullOrEmpty(subComponentFileName))
+                        {
+                            var componentsJSONPath = Path.Combine(enlistmentRoot, ".corext", "Configs", subComponentFileName);
+                            var jDoc = GetJsonDocumentForComponentsFile(componentsJSONPath);
+
+                            if (jDoc != null && !ComponentFileToDocumentMap.ContainsKey(componentsJSONPath))
+                            {
+                                ComponentFileToDocumentMap[componentsJSONPath] = jDoc;
+                                PopulateComponentToFileMapForFile(jDoc, componentsJSONPath);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void PopulateComponentToFileMapForFile(JObject jDocument, string componentsJsonFileName)
+        {
+            if (jDocument != null && !string.IsNullOrEmpty(componentsJsonFileName))
+            {
+                var jComponents = (JObject)jDocument["Components"];
+
+                if (jComponents != null)
+                {
+                    Dictionary<string, JToken> componentsMap = jComponents.ToObject<Dictionary<string, JToken>>();
+
+                    if (componentsMap != null && componentsMap.Any())
+                    {
+                        foreach (var kvp in componentsMap)
+                        {
+                            if (!ComponentToFileMap.ContainsKey(kvp.Key))
+                            {
+                                ComponentToFileMap[kvp.Key] = componentsJsonFileName;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static JObject GetJsonDocumentForComponentsFile(string componentsJSONPath)
+        {
+            JObject jsonDocument = null;
+
+            if (File.Exists(componentsJSONPath))
```

Comment:
Wonder if we should delete this. At this point, we don't do automated insertions for Dev14 so we should always have one.

Reply:
Ideally yes. You are right.
But I think its good to have to keep the tool robust. This function will now get called for every distributed component.json file it sees in the main components.json. In case due to any issue if any file is missing, it will not crash.